--- /tmp/stringi-bbbf25bbe6e1aa198389dd08636a3360b5788213/src/icu69/common/rbbi_cache.cpp	2021-05-01 15:37:10.000000000 +1000
+++ common/rbbi_cache.cpp	2021-05-23 17:51:43.986745303 +1000
@@ -75,6 +75,9 @@
         }
     }
     UPRV_UNREACHABLE;
+#ifdef U_STRINGI_PATCHES
+    return TRUE;
+#endif
 }
 
 
@@ -115,6 +118,9 @@
         }
     }
     UPRV_UNREACHABLE;
+#ifdef U_STRINGI_PATCHES
+    return TRUE;
+#endif
 }
 
 void RuleBasedBreakIterator::DictionaryCache::populateDictionary(int32_t startPos, int32_t endPos,
