--- /tmp/stringi-bbbf25bbe6e1aa198389dd08636a3360b5788213/src/icu69/common/uassert.h	2021-05-01 15:37:10.000000000 +1000
+++ common/uassert.h	2023-01-08 11:23:55.924725772 +1100
@@ -45,7 +45,21 @@
 #if defined(UPRV_UNREACHABLE)
     // Use the predefined value.
 #else
-#   define UPRV_UNREACHABLE abort()
+#ifdef U_STRINGI_PATCHES
+// Marek
+
+#if defined(__GNUC__) && __GNUC__ >= 3
+#define NORET __attribute__((noreturn))
+#else
+#define NORET
 #endif
 
+extern "C" void NORET Rf_error(const char *, ...);
+
+#   define UPRV_UNREACHABLE (Rf_error("Internal error in ICU: UPRV_UNREACHABLE"))
+
+#else /* !U_STRINGI_PATCHES */
+#   define UPRV_UNREACHABLE abort()
+#endif
+#endif /* U_STRINGI_PATCHES */
 #endif
