--- /tmp/stringi-bbbf25bbe6e1aa198389dd08636a3360b5788213/src/icu69/common/ubiditransform.cpp	2021-05-01 15:37:10.000000000 +1000
+++ common/ubiditransform.cpp	2023-01-16 19:35:27.070820544 +1100
@@ -394,8 +394,13 @@
         case UBIDI_DEFAULT_LTR:
         case UBIDI_DEFAULT_RTL: {
             UBiDiLevel level = static_cast<UBiDiLevel>(ubidi_getBaseDirection(text, length));
+#ifdef U_STRINGI_PATCHES
+            *pInLevel = static_cast<UBiDiLevel>(level != UBIDI_NEUTRAL) ? level
+                    : ((*pInLevel == UBIDI_DEFAULT_RTL) ? static_cast<UBiDiLevel>(RTL) : static_cast<UBiDiLevel>(LTR));
+#else /* !U_STRINGI_PATCHES */
             *pInLevel = static_cast<UBiDiLevel>(level != UBIDI_NEUTRAL) ? level
                     : *pInLevel == UBIDI_DEFAULT_RTL ? static_cast<UBiDiLevel>(RTL) : static_cast<UBiDiLevel>(LTR);
+#endif /* U_STRINGI_PATCHES */
             break;
         }
         default:
@@ -498,8 +503,8 @@
 
     /* Checking for U_SUCCESS() within the loop to bail out on first failure. */
     for (action = pBiDiTransform->pActiveScheme->actions; *action && U_SUCCESS(*pErrorCode); action++) {
-        if ((*action)(pBiDiTransform, pErrorCode)) {
-            if (action + 1) {
+        if (action && (*action)(pBiDiTransform, pErrorCode)) {
+            if (*(action + 1)) {  // Marek's patch
                 updateSrc(pBiDiTransform, pBiDiTransform->dest, *pBiDiTransform->pDestLength,
                         *pBiDiTransform->pDestLength, pErrorCode);
             }
