--- /tmp/stringi-bbbf25bbe6e1aa198389dd08636a3360b5788213/src/icu69/common/utrace.cpp	2021-05-01 15:37:10.000000000 +1000
+++ common/utrace.cpp	2021-05-23 17:51:44.078744403 +1000
@@ -48,7 +48,7 @@
 utrace_exit(int32_t fnNumber, int32_t returnType, ...) {
     if (pTraceExitFunc != NULL) {
         va_list     args;
-        const char *fmt;
+        const char *fmt = nullptr;
 
         switch (returnType) {
         case 0:
