--- /tmp/stringi-bbbf25bbe6e1aa198389dd08636a3360b5788213/src/icu69/i18n/calendar.cpp	2021-05-01 15:37:10.000000000 +1000
+++ i18n/calendar.cpp	2023-01-08 11:23:55.924725772 +1100
@@ -840,10 +840,17 @@
         fWeekendCease            = right.fWeekendCease;
         fWeekendCeaseMillis      = right.fWeekendCeaseMillis;
         fNextStamp               = right.fNextStamp;
+#ifdef U_STRINGI_PATCHES
+        // fix for an annoying compiler warning (false positive)...
+        // snprintf writes at most *size* bytes (including the terminating null byte *str* (1st arg)
+        snprintf((char*)validLocale, ULOC_FULLNAME_CAPACITY, "%s", (char*)right.validLocale);
+        snprintf((char*)actualLocale, ULOC_FULLNAME_CAPACITY, "%s", (char*)right.actualLocale);
+#else //!U_STRINGI_PATCHES
         uprv_strncpy(validLocale, right.validLocale, sizeof(validLocale));
         uprv_strncpy(actualLocale, right.actualLocale, sizeof(actualLocale));
         validLocale[sizeof(validLocale)-1] = 0;
         actualLocale[sizeof(validLocale)-1] = 0;
+#endif // U_STRINGI_PATCHES
     }
 
     return *this;
