--- /tmp/stringi-bbbf25bbe6e1aa198389dd08636a3360b5788213/src/icu69/i18n/decNumber.h	2021-05-01 15:37:10.000000000 +1000
+++ i18n/decNumber.h	2021-05-23 17:51:44.242742791 +1000
@@ -64,7 +64,11 @@
   /* that the structure will be immediately followed by extra space,  */
   /* as required.  DECNUMDIGITS is always >0.                         */
   #if !defined(DECNUMDIGITS)
+#ifdef U_STRINGI_PATCHES
+    #define DECNUMDIGITS 4
+#else /* !U_STRINGI_PATCHES */
     #define DECNUMDIGITS 1
+#endif /* U_STRINGI_PATCHES */
   #endif
 
   /* The size (integer data type) of each unit is determined by the   */
