--- /tmp/stringi-bbbf25bbe6e1aa198389dd08636a3360b5788213/src/icu69/i18n/islamcal.cpp	2021-05-01 15:37:10.000000000 +1000
+++ i18n/islamcal.cpp	2021-05-23 17:51:44.262742594 +1000
@@ -604,8 +604,8 @@
 * @draft ICU 2.4
 */
 void IslamicCalendar::handleComputeFields(int32_t julianDay, UErrorCode &status) {
-    int32_t year, month, dayOfMonth, dayOfYear;
-    int32_t startDate;
+    int32_t year=0, month=0, dayOfMonth=0, dayOfYear=0;
+    int32_t startDate=0;
     int32_t days = julianDay - CIVIL_EPOC;
 
     if (cType == CIVIL || cType == TBLA) {
