--- /tmp/stringi-bbbf25bbe6e1aa198389dd08636a3360b5788213/src/icu69/i18n/number_patternmodifier.cpp	2021-05-01 15:37:10.000000000 +1000
+++ i18n/number_patternmodifier.cpp	2021-05-23 17:51:44.286742358 +1000
@@ -245,6 +245,9 @@
     (void)field;
     // This method is not currently used.
     UPRV_UNREACHABLE;
+#ifdef U_STRINGI_PATCHES
+    return false;
+#endif
 }
 
 void MutablePatternModifier::getParameters(Parameters& output) const {
@@ -257,6 +260,9 @@
     (void)other;
     // This method is not currently used.
     UPRV_UNREACHABLE;
+#ifdef U_STRINGI_PATCHES
+    return false;
+#endif
 }
 
 int32_t MutablePatternModifier::insertPrefix(FormattedStringBuilder& sb, int position, UErrorCode& status) {
@@ -325,12 +331,18 @@
             return UnicodeString(u"\uFFFD");
         default:
             UPRV_UNREACHABLE;
+#ifdef U_STRINGI_PATCHES
+    return UnicodeString(u"\uFFFD");
+#endif
     }
 }
 
 UnicodeString MutablePatternModifier::toUnicodeString() const {
     // Never called by AffixUtils
     UPRV_UNREACHABLE;
+#ifdef U_STRINGI_PATCHES
+    return UnicodeString(u"\uFFFD");
+#endif
 }
 
 #endif /* #if !UCONFIG_NO_FORMATTING */
