--- /tmp/stringi-bbbf25bbe6e1aa198389dd08636a3360b5788213/src/icu69/i18n/number_modifiers.cpp	2021-05-01 15:37:10.000000000 +1000
+++ i18n/number_modifiers.cpp	2021-05-23 17:51:44.286742358 +1000
@@ -93,6 +93,9 @@
     (void)field;
     // This method is not currently used.
     UPRV_UNREACHABLE;
+#ifdef U_STRINGI_PATCHES
+    return false;
+#endif
 }
 
 void ConstantAffixModifier::getParameters(Parameters& output) const {
@@ -182,6 +185,9 @@
     (void)field;
     // This method is not currently used.
     UPRV_UNREACHABLE;
+#ifdef U_STRINGI_PATCHES
+    return false;
+#endif
 }
 
 void SimpleModifier::getParameters(Parameters& output) const {
