--- /tmp/stringi-bbbf25bbe6e1aa198389dd08636a3360b5788213/src/icu69/i18n/number_rounding.cpp	2021-05-01 15:37:10.000000000 +1000
+++ i18n/number_rounding.cpp	2021-05-23 17:51:44.286742358 +1000
@@ -294,6 +294,9 @@
     settings.fMaxFrac = static_cast<digits_t>(maxFrac);
     settings.fMinSig = -1;
     settings.fMaxSig = -1;
+#ifdef U_STRINGI_PATCHES
+    settings.fPriority = UNUM_ROUNDING_PRIORITY_RELAXED;
+#endif
     PrecisionUnion union_;
     union_.fracSig = settings;
     return {RND_FRACTION, union_};
@@ -305,6 +308,9 @@
     settings.fMaxFrac = -1;
     settings.fMinSig = static_cast<digits_t>(minSig);
     settings.fMaxSig = static_cast<digits_t>(maxSig);
+#ifdef U_STRINGI_PATCHES
+    settings.fPriority = UNUM_ROUNDING_PRIORITY_RELAXED;
+#endif
     PrecisionUnion union_;
     union_.fracSig = settings;
     return {RND_SIGNIFICANT, union_};
