--- /tmp/stringi-bbbf25bbe6e1aa198389dd08636a3360b5788213/src/icu69/i18n/number_scientific.cpp	2021-05-01 15:37:10.000000000 +1000
+++ i18n/number_scientific.cpp	2021-05-23 17:51:44.290742319 +1000
@@ -97,6 +97,9 @@
     (void)field;
     // This method is not used for inner modifiers.
     UPRV_UNREACHABLE;
+#ifdef U_STRINGI_PATCHES
+    return false;
+#endif
 }
 
 void ScientificModifier::getParameters(Parameters& output) const {
