--- /tmp/stringi-bbbf25bbe6e1aa198389dd08636a3360b5788213/src/icu69/i18n/regexcmp.cpp	2021-05-01 15:37:10.000000000 +1000
+++ i18n/regexcmp.cpp	2021-05-23 17:51:44.310742123 +1000
@@ -3129,6 +3129,9 @@
         case URX_LBN_END:
             UPRV_UNREACHABLE;     // Shouldn't get here.  These ops should be
                                  //  consumed by the scan in URX_LA_START and LB_START
+#ifdef U_STRINGI_PATCHES
+            break;
+#endif
         default:
             UPRV_UNREACHABLE;
             }
@@ -3697,6 +3700,9 @@
             // These opcodes will be skipped over by code for URX_CTR_INIT.
             // We shouldn't encounter them here.
             UPRV_UNREACHABLE;
+#ifdef U_STRINGI_PATCHES
+            break;
+#endif
 
         case URX_LOOP_SR_I:
         case URX_LOOP_DOT_I:
