--- /tmp/stringi-bbbf25bbe6e1aa198389dd08636a3360b5788213/src/icu69/i18n/rematch.cpp	2021-05-01 15:37:10.000000000 +1000
+++ i18n/rematch.cpp	2021-05-23 17:51:44.318742044 +1000
@@ -720,6 +720,9 @@
                 return FALSE;
         }
         UPRV_UNREACHABLE;
+#ifdef U_STRINGI_PATCHES
+            break;
+#endif
 
     case START_START:
         // Matches are only possible at the start of the input string
@@ -768,6 +771,9 @@
             }
         }
         UPRV_UNREACHABLE;
+#ifdef U_STRINGI_PATCHES
+            break;
+#endif
 
     case START_STRING:
     case START_CHAR:
@@ -800,6 +806,9 @@
            }
         }
         UPRV_UNREACHABLE;
+#ifdef U_STRINGI_PATCHES
+            break;
+#endif
 
     case START_LINE:
         {
@@ -883,6 +892,9 @@
     }
 
     UPRV_UNREACHABLE;
+#ifdef U_STRINGI_PATCHES
+    return FALSE;
+#endif
 }
 
 
@@ -994,6 +1006,9 @@
                 return FALSE;
         }
         UPRV_UNREACHABLE;
+#ifdef U_STRINGI_PATCHES
+        return FALSE;
+#endif
 
     case START_START:
         // Matches are only possible at the start of the input string
@@ -1036,6 +1051,9 @@
         }
     }
     UPRV_UNREACHABLE;
+#ifdef U_STRINGI_PATCHES
+    break;
+#endif
 
     case START_STRING:
     case START_CHAR:
@@ -1065,6 +1083,9 @@
         }
     }
     UPRV_UNREACHABLE;
+#ifdef U_STRINGI_PATCHES
+        return FALSE;
+#endif
 
     case START_LINE:
     {
@@ -1139,6 +1160,9 @@
     }
 
     UPRV_UNREACHABLE;
+#ifdef U_STRINGI_PATCHES
+    return FALSE;
+#endif
 }
 
 
