--- /tmp/stringi-bbbf25bbe6e1aa198389dd08636a3360b5788213/src/icu69/i18n/ucal.cpp	2021-05-01 15:37:10.000000000 +1000
+++ i18n/ucal.cpp	2021-05-23 17:51:44.338741847 +1000
@@ -166,7 +166,11 @@
   }
 
   if ( caltype == UCAL_GREGORIAN ) {
+#ifdef U_STRINGI_PATCHES
+      char localeBuf[ULOC_LOCALE_IDENTIFIER_CAPACITY+1];
+#else /* !U_STRINGI_PATCHES */
       char localeBuf[ULOC_LOCALE_IDENTIFIER_CAPACITY];
+#endif /* U_STRINGI_PATCHES */
       if ( locale == nullptr ) {
           locale = uloc_getDefault();
       }
