--- /tmp/stringi-bbbf25bbe6e1aa198389dd08636a3360b5788213/src/icu69/i18n/uspoof_impl.cpp	2021-05-01 15:37:10.000000000 +1000
+++ i18n/uspoof_impl.cpp	2023-01-16 19:35:27.074820720 +1100
@@ -945,7 +945,7 @@
     uint32_t magic = ds->readUInt32(spoofDH->fMagic);
     ds->writeUInt32((uint32_t *)&outputDH->fMagic, magic);
 
-    if (outputDH->fFormatVersion != spoofDH->fFormatVersion) {
+    if ((uint8_t*)(outputDH->fFormatVersion) != (uint8_t*)(spoofDH->fFormatVersion)) {  // Marek's patch
         uprv_memcpy(outputDH->fFormatVersion, spoofDH->fFormatVersion, sizeof(spoofDH->fFormatVersion));
     }
     // swap starting at fLength
