--- /tmp/stringi-d268f16043fb6de40901e2eab31b39f1ecdcd3f3/src/icu61/i18n/decNumber.h	2018-04-21 00:45:30.000000000 +1000
+++ /home/gagolews/R/stringi/src/icu61/i18n/decNumber.h	2019-12-21 22:47:29.654679323 +1100
@@ -61,10 +61,11 @@
 
   /* DECNUMDIGITS is the default number of digits that can be held in */
   /* the structure.  If undefined, 1 is assumed and it is assumed     */
+  // @mod Marek/stringi 1→4
   /* that the structure will be immediately followed by extra space,  */
   /* as required.  DECNUMDIGITS is always >0.                         */
   #if !defined(DECNUMDIGITS)
-    #define DECNUMDIGITS 1
+    #define DECNUMDIGITS 4
   #endif
 
   /* The size (integer data type) of each unit is determined by the   */
