--- /tmp/stringi-d268f16043fb6de40901e2eab31b39f1ecdcd3f3/src/icu61/i18n/ucal.cpp	2018-04-21 00:45:30.000000000 +1000
+++ /home/gagolews/R/stringi/src/icu61/i18n/ucal.cpp	2019-02-15 20:26:07.348449431 +1100
@@ -148,7 +148,7 @@
   }
 
   if ( caltype == UCAL_GREGORIAN ) {
-      char  localeBuf[ULOC_LOCALE_IDENTIFIER_CAPACITY];
+      char  localeBuf[ULOC_LOCALE_IDENTIFIER_CAPACITY+1];
       if ( locale == NULL ) {
           locale = uloc_getDefault();
       }
