--- /tmp/stringi-d268f16043fb6de40901e2eab31b39f1ecdcd3f3/src/icu61/common/rbbitblb.cpp	2018-04-21 00:45:30.000000000 +1000
+++ /home/gagolews/R/stringi/src/icu61/common/rbbitblb.cpp	2019-02-15 20:26:07.344449431 +1100
@@ -1085,8 +1085,8 @@
     int32_t numStates = fDStates->size();
     int32_t numCols = fRB->fSetBuilder->getNumCharCategories();
 
-    uint16_t table_base;
-    uint16_t table_dupl;
+    uint16_t table_base = 0;
+    uint16_t table_dupl = 0;
     for (; baseCategory < numCols-1; ++baseCategory) {
         for (duplCategory=baseCategory+1; duplCategory < numCols; ++duplCategory) {
              for (int32_t state=0; state<numStates; state++) {
