--- /tmp/stringi-d268f16043fb6de40901e2eab31b39f1ecdcd3f3/src/icu61/common/ubiditransform.cpp	2018-04-21 00:45:30.000000000 +1000
+++ /home/gagolews/R/stringi/src/icu61/common/ubiditransform.cpp	2019-02-15 20:26:07.348449431 +1100
@@ -394,8 +394,8 @@
         case UBIDI_DEFAULT_LTR:
         case UBIDI_DEFAULT_RTL: {
             UBiDiLevel level = ubidi_getBaseDirection(text, length);
-            *pInLevel = level != UBIDI_NEUTRAL ? level
-                    : *pInLevel == UBIDI_DEFAULT_RTL ? RTL : LTR;
+            *pInLevel = ((level != UBIDI_NEUTRAL) ? level
+                    : ((*pInLevel == UBIDI_DEFAULT_RTL) ? RTL : LTR));
             break;
         }
         default:
