--- /tmp/stringi-bbbf25bbe6e1aa198389dd08636a3360b5788213/src/icu69/unicode/uidna.h	2021-05-01 15:37:10.000000000 +1000
+++ unicode/uidna.h	2021-05-23 17:51:44.414741100 +1000
@@ -576,7 +576,7 @@
  *         in which case it will be greater than destCapacity.
  * @deprecated ICU 55 Use UTS #46 instead via uidna_openUTS46() or class IDNA.
  */
-U_DEPRECATED int32_t U_EXPORT2
+U_CAPI int32_t U_EXPORT2
 uidna_toASCII(const UChar* src, int32_t srcLength, 
               UChar* dest, int32_t destCapacity,
               int32_t options,
