--- /tmp/stringi-d268f16043fb6de40901e2eab31b39f1ecdcd3f3/src/icu61/unicode/umachine.h	2018-04-21 00:45:30.000000000 +1000
+++ /home/gagolews/R/stringi/src/icu61/unicode/umachine.h	2019-02-15 20:26:07.352449432 +1100
@@ -112,7 +112,8 @@
 /** This is used to declare a function as a draft public ICU C API  */
 #define U_DRAFT  U_CAPI
 /** This is used to declare a function as a deprecated public ICU C API  */
-#define U_DEPRECATED U_CAPI U_ATTRIBUTE_DEPRECATED
+#define U_DEPRECATED U_CAPI
+//#define U_DEPRECATED U_CAPI U_ATTRIBUTE_DEPRECATED
 /** This is used to declare a function as an obsolete public ICU C API  */
 #define U_OBSOLETE U_CAPI
 /** This is used to declare a function as an internal ICU C API  */
